galsim/lsst/lsst_wcs.py

```diff
@@ -0,0 +1,1153 @@
+        self._cos_rot = np.cos(self._rotation_angle/galsim.radians)
+        self._sin_rot = np.sin(self._rotation_angle/galsim.radians)
+        self._cos_dec = np.cos(self._pointing.dec/galsim.radians)
+        self._sin_dec = np.sin(self._pointing.dec/galsim.radians)
+
+
+    @property
+    def pointing(self):
+        """
+        A galsim.CelestialCoord characterizing the direction the camera is pointing
+        """
+        return self._pointing
+
+
+    @property
+    def rotation_angle(self):
+        """
+        A galsim.Angle object characterizing the rotation of the camera with respect
+        to the sky.
+
+        rotation_angle = 0 degrees means north is in the +x direction (in pixel coordinates)
+                         and east is in the +y direction
+
+        rotation_angle = 90 degrees means north is +y and east is -x
+
+        rotation_angle = -90 degrees means north is -y and east is +x
+
+        rotation_angle = 180 degrees means north is -x and east is -y
+        """
+        return self._rotation_angle
+
+
+    def pupilCoordsFromPoint(self, point):
+        """
+        Convert from RA, Dec into coordinates on the pupil
+
+        inputs
+        ------------
+        point is a CelestialCoord (or a list of CelestialCoords) indicating
+        the positions to be transformed
+
+        outputs
+        ------------
+        The x and y coordinates on the pupil in radians
+        """
+
+        if not hasattr(point, '__len__'):
+            pt = self._pointing.project(point, projection='gnomonic')
+            x = pt.x
+            y = pt.y
+        else:
+            x = []
+            y = []
+            for pp in point:
+                pt = self._pointing.project(pp, projection='gnomonic')
+                x.append(pt.x)
+                y.append(pt.y)
+            x = np.array(x)
+            y = np.array(y)
+
+        return (x*self._cos_rot - y*self._sin_rot)*galsim.arcsec, \
+               (x*self._sin_rot + y*self._cos_rot)*galsim.arcsec
+
+
+    def pupilCoordsFromFloat(self, ra, dec):
+        """
+        Convert from RA, Dec into coordinates on the pupil
+
+        Note: this method just reimplements the PALPY method Ds2tp
+        with some minor adjustments for sign conventions.
+
+        inputs
+        ------------
+        ra is in radians.  Can be a list.
+
+        dec is in radians.  Can be a list.
+
+        outputs
+        ------------
+        The x and y coordinates on the pupil in radians
+        """
+
+        dra = ra - self._pointing.ra/galsim.radians
+        cradif = np.cos(dra)
+        sradif = np.sin(dra)
+        sdec = np.sin(dec)
+        cdec = np.cos(dec)
+        denom = sdec * self._sin_dec + cdec * self._cos_dec * cradif
+        xx = cdec * sradif/denom
+        yy = (sdec * self._cos_dec - cdec * self._sin_dec * cradif)/denom
+        xx *= -1.0
+        return xx*self._cos_rot - yy*self._sin_rot, xx*self._sin_rot + yy*self._cos_rot
```

Comment:
This seems like a reimplementation of the above function `pupilCoordsFromPoint`.  Seems like both should be simple wrappers around something that does the actual work and would be a single implementation.  Also, I suspect you'll want to use the `project_rad` function, which acts like `project` (used above) but on numpy arrays of ra, dec.